src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -258,4 +279,41 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
       postResponse.entity.asString.contains(Conflict.reason)
   }
 
+  // get a mapping of FireCloud user name to NIH User name, for only those Thurloe users with a non-expired NIH link
+  def getCurrentNihUsernameMap: Future[Map[String, String]] = {
+
+    val pipeline = addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken)) ~> sendReceive
+    def getAllUserValuesForKey(key: String): Future[Map[String, String]] = pipeline {
+      Get(s"${UserService.remoteGetQueryURL}key=${key}")
+    } map { response =>
+      if (response.status != OK) throw new Exception(response.toString)   // TODO: better error handling
```

Comment:
nope, nothing standard - use an ErrorReport if you can, or create a new Exception class.